src/tasks/components/TaskCard.tsx

```diff
@@ -138,61 +143,91 @@ export class TaskCard extends PureComponent<
-              disabled={isPinned}
-            />
-          </Context.Menu>
-        )}
-        <Context.Menu
-          icon={IconFont.Trash}
-          color={ComponentColor.Danger}
-          testID="context-delete-menu"
-        >
-          <Context.Item
-            label="Delete"
-            action={this.handleOnDelete}
-            value={task}
-            testID="context-delete-task"
-          />
-        </Context.Menu>
-      </Context>
+      <FlexBox margin={ComponentSize.ExtraSmall}>
+        <ConfirmationButton
+          color={ComponentColor.Colorless}
+          icon={IconFont.Trash_New}
+          shape={ButtonShape.Square}
+          size={ComponentSize.ExtraSmall}
+          confirmationLabel="Yes, Delete this task"
+          onConfirm={() => {
+            this.handleOnDelete(task)
+          }}
+          confirmationButtonText="Confirm"
+          testID={`context-delete-menu ${task.name}`}
+        />
+        <SquareButton
+          ref={settingsRef}
+          size={ComponentSize.ExtraSmall}
+          icon={IconFont.CogSolid_New}
+          color={ComponentColor.Colorless}
+          testID="context-menu-task"
+        />
+        <Popover
+          appearance={Appearance.Outline}
+          enableDefaultStyles={false}
+          style={{minWidth: '112px'}}
+          contents={() => (
+            <List>
+              <List.Item
+                onClick={this.handleExport}
+                size={ComponentSize.Small}
+                style={{fontWeight: 500}}
+                testID="context-export-task"
+              >
+                Export
+              </List.Item>
+              <List.Item
+                onClick={this.handleEditTask}
+                size={ComponentSize.Small}
+                style={{fontWeight: 500}}
+                testID="context-edit-task"
+              >
+                Edit
+              </List.Item>
+              <List.Item
+                value={task.id}
+                onClick={this.handleRunTask}
+                size={ComponentSize.Small}
+                style={{fontWeight: 500}}
+                testID="context-run-task"
+              >
+                Run
+              </List.Item>
+              <List.Item
+                value={task}
+                onClick={onClone}
+                size={ComponentSize.Small}
+                style={{fontWeight: 500}}
+                testID="context-clone-task"
+              >
+                Clone
+              </List.Item>
+              {isFlagEnabled('pinnedItems') && CLOUD && (
+                <List.Item
+                  onClick={this.handlePinTask}
+                  disabled={isPinned}
+                  size={ComponentSize.Small}
+                  style={{fontWeight: 500}}
+                  testID="context-pin-task"
+                >
+                  Pin
+                </List.Item>
+              )}
+            </List>
+          )}
+          triggerRef={settingsRef}
+        />
+      </FlexBox>
```

Comment:
love the removal of that `ContextMenu` and its included `ClickOutside` component